### PR TITLE
latexified figure book1/4.24 

### PR DESCRIPTION
**Pull request by shobro**
## Description

Latexified and added file [samplingDistributionGaussianShrinkage.ipynb](https://github.com/probml/pyprobml/blob/master/notebooks/book1/04/samplingDistributionGaussianShrinkage.ipynb)

### Figure Number

`book1/4.24`
### Figures
|Before PR|After PR|
|--------------|--------|
|beforePR|Screenshot from 2022-06-02 01-26-28|

### Issue 
#34 

## Steps

- [x] Step1 - latexified the figures.
- [x] Step2 - jaxified the code.
- [x] Step3 - added the interactive demo which demonstrates changes in posterior mean with change in no of samples (n)

## Checklist:

- [x] Performed a self-review of the code
- [x] Tested on Google Colab.

### CODE REVIEW

**Comment by shobro** [comment]
@karm-patel I think I understood the point you were trying to make in #45 . I have update the [gist](https://gist.github.com/shobro/9e10084de4e6222d2be3675bca17d91e) accordingly. Please have a look.

**Comment by karm-patel** [comment]
Set `fig_height` = 1.85, I checked in latex source code

**Comment by shobro** [comment]
updated the [gist](https://gist.github.com/shobro/9e10084de4e6222d2be3675bca17d91e)

**Comment by karm-patel** [comment]
Ideally, we should use the `default font_size = 9`, observe with this smaller `font_size = 3.5` , it will be difficult to read text for reader

**Comment by shobro** [comment]
@karm-patel  I tried with various font sizes best I can fit the text without overlapping it with other parts of the graph is either font _size = 4 or 4.5.

I have shared various latexified figure with various font sizes [here](https://gist.github.com/shobro/f52c8f197ea79c2dd51c221e4f7c8c57)

**Comment by shobro** [comment]
@karm-patel Earlier I made some changes in the legend to improve the understandability of the graph which made the legend lengthy. I think those changes are not necessary. 

Currently, I have shortened the legend text as a result we can increase the font size to 6 and improve readability. I made a gist [here](https://gist.github.com/shobro/8bc886f7a4e6a65f559a85bbf47ab93b) to show you the change.

**Comment by karm-patel** [comment]
I think still you can increase the font size, you just need some workarounds
1. If title of 1st subfigure overflow. then you can break down the title into two lines and see how it looks.  
2. Problem in legend of the figure can be solved by dividing to two legends, see this: https://gist.github.com/karm-patel/15b1e1895756088725872bba9204c9d1
3. And you should write the legend label as `postMean (k_0 = 1)` , the current one postMean0 looks ambiguous.

**Comment by shobro** [comment]
@karm-patel I made changes according to your suggestions please have a look [here](https://gist.github.com/shobro/dba96e58128e9560c73dada5891d3e62).

**Comment by karm-patel** [comment]
@shobro, Have you updated the plot as per my suggestions?

**Comment by shobro** [comment]
@karm-patel I have updated changes as per your suggestions please have a look

File: notebooks/book1/04/samplingDistributionGaussianShrinkage.ipynb [hunk 1]
```diff
@@ -0,0 +1,566 @@
+{
```
Comment (karm-patel) [inline notebooks/book1/04/samplingDistributionGaussianShrinkage.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #11.    latexify(width_scale_factor=2, fig_height=1.85, font_size=7)
</pre><p>Don't decrease font_size of all labels, decrease only those labels which are overflowing, contact anand how to do this, I explained him once</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/46/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

**Comment by karm-patel** [comment]
Any updates on this?

File: notebooks/book1/04/samplingDistributionGaussianShrinkage.ipynb [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
```
Comment (nipunbatra) [inline notebooks/book1/04/samplingDistributionGaussianShrinkage.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.        logp = -0.5 * jnp.sum(X * X / Sigma, axis=1)
</pre><p>Just curious -- would it make sense to use tfp.distributions and get the logprob through that? It becomes easier to comprehend the code that way. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/46/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (nipunbatra) [inline notebooks/book1/04/samplingDistributionGaussianShrinkage.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #10.        k0s = jnp.arange(k0)
</pre><p>What are k0s? and k0? It may make sense to add some comments for a future reader</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/46/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

**Comment by karm-patel** [comment]
Okay, this looks good to me, you can create a  PR on main repo.
